### PR TITLE
feat: add download count field and unit testing for attachment.

### PR DESCRIPTION
**Pull request by appleboy**
* Add download count field in `attachment` table. See [GitHub API](https://developer.github.com/v3/repos/releases/#get-a-single-release-asset)
* Add unit testing.

### CODE REVIEW

**Comment by strk** [comment]
It takes a migration to add that new field, right ?

**Comment by appleboy** [comment]
@strk Yes.

File: models/attachment_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIncreaseDownloadCount(t *testing.T) {
+	assert.NoError(t, PrepareTestDatabase())
+
+	attachment, err := GetAttachmentByUUID("1234567890")
+	assert.NoError(t, err)
```
Comment (bkcsoft) [inline models/attachment_test.go:17]: `assert.Equal(t, attachment.DownloadCount, int64(0))` ?

**Review by bkcsoft — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: models/attachment_test.go (hunk 1 above)
Comment (appleboy) [reply models/attachment_test.go:17]: @bkcsoft Done!

Comment (appleboy) [reply models/attachment_test.go:17]: See https://github.com/go-gitea/gitea/pull/1512/commits/8278ac976780fff9a42d2e1bbffba37a480b55e5

**Review by bkcsoft — APPROVED** [review]

**Comment by bkcsoft** [comment]
LGTM

**Comment by strk** [comment]
isn't this still lacking a migration ?

File: models/attachment.go [hunk 2]
```diff
@@ -20,15 +20,15 @@ import (
...
-
-	Created     time.Time `xorm:"-"`
-	CreatedUnix int64
+	ID            int64  `xorm:"pk autoincr"`
+	UUID          string `xorm:"uuid UNIQUE"`
+	IssueID       int64  `xorm:"INDEX"`
+	CommentID     int64
+	ReleaseID     int64 `xorm:"INDEX"`
+	Name          string
+	DownloadCount int64
```
Comment (bkcsoft) [inline models/attachment.go]: Instead of migration, do this:
```go
  DownloadCount int64 `xorm:"DEFAULT 0"`
```

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: models/attachment.go (hunk 2 above)
Comment (appleboy) [reply models/attachment.go]: Done https://github.com/go-gitea/gitea/pull/1512/commits/e35bbb2f1b2e195ab02de1c606367056e486de76

Comment (strk) [reply models/attachment.go]: I remember @lunny calling for always adding a migration, not sure why

Comment (lunny) [reply models/attachment.go]: It's better to do that. I think we can begin to ask all schema changes should use https://github.com/go-xorm/xorm/tree/master/migrate

**Comment by lunny** [comment]
LGTM

**Pull request merged**